packages/ns-api/files/ns.threatshield

```diff
@@ -541,6 +543,138 @@ def dns_delete_bypass(e_uci, payload):
+                                    capture_output=True, text=True)
+        data = json.loads(output_ipv6.stdout)
+    except subprocess.CalledProcessError:
+        data = {}
+
+    blocks = blocks + __parse_set(data)
+    blocks = filter(lambda x: isinstance(x, dict) and "val" in x, blocks)
+    blocks = sorted(blocks, key=lambda x: x["timeout"] - x["expires"])
+    response = []
+    for block in blocks:
+        if not isinstance(block["val"], list):
+            block["val"] = [block["val"]]
+        for item in block["val"]:
+            if isinstance(item, ipaddress.IPv4Address) or isinstance(item, ipaddress.IPv6Address):
+                response.append({
+                    "ip": str(item),
+                    "timeout": block["timeout"],
+                    "expires": block["expires"],
+                })
+            if isinstance(item, ipaddress.IPv4Network) or isinstance(item, ipaddress.IPv6Network):
+                for addr in item:
+                    if isinstance(addr, ipaddress.IPv4Address) or isinstance(addr, ipaddress.IPv6Address):
+                        response.append({
+                            "ip": str(addr),
+                            "timeout": block["timeout"],
+                            "expires": block["expires"],
+                        })
+
+    return response
+
+
+def remove_active_block():
+    data = json.loads(sys.stdin.read())
+    if data.get("ip", "") == "":
+        return validation_error("ip", "required")
+
+    try:
+        ip = ipaddress.ip_address(data["ip"])
+    except ValueError:
+        return validation_error("ip", "invalid", data["ip"])
+
+    blocks = filter(lambda x: x["ip"] == str(ip), list_active_blocks())
+    found = len(list(blocks)) > 0
+    if not found:
+        return validation_error("ip", "invalid", str(ip))
+
+    if isinstance(ip, ipaddress.IPv4Address):
+        blocklist = "blocklistv4"
+    else:
+        blocklist = "blocklistv6"
+
+    try:
+        subprocess.run(["nft", "delete", "element", "inet", "banIP", blocklist, "{" + str(ip) + "}"], check=True)
+    except subprocess.CalledProcessError:
+        return generic_error("remove_active_block_failed")
+
+    return {'message': 'success'}
+
+
+def is_ip_blocked():
+    data = json.loads(sys.stdin.read())
+    if "ip" not in data or data["ip"] == "":
+        return validation_error("ip", "required")
+    try:
+        result = subprocess.run(["/etc/init.d/banip", "search", data["ip"]], check=True, capture_output=True, text=True)
+        matches = re.findall(r"Set '([^']+)'", result.stdout)
+        matches = [match for match in matches if not match.startswith("allow")]
+
+        if len(matches) > 0:
+            return {
+                'found': True,
+                'list': matches[0]
+            }
+    except subprocess.CalledProcessError:
+        return validation_error("banip", "processing")
+
+    return {
+        'found': False
```

Comment:
Do not know if it's better for the UI, or not, but I'd add always add the `list` field to have a uniform return value
```suggestion
        'found': False
        'list': []
```

Reply:
UI takes care of that, without issues, even with typing support: https://github.com/NethServer/nethsecurity-ui/pull/538/files#diff-cf73b5e916d8da4cf7120ea74aefe5eaf0db51e79c3a0ec0fc2a1b7cf03247a7R21